azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -736,32 +731,159 @@ def do_retry():
 ]
```

Comment:
Could we move this inside the ReconnectState class, since that's the only place it's used? I don't like having global definitions hanging around.